### PR TITLE
Logger updates

### PR DESCRIPTION
**Pull request by jithinjosepkl**
Added function name, line number info to logger output

### CODE REVIEW

**Comment by goodell** [comment]
Looks mostly OK to me.  I didn't originally add this feature because I thought it might end up cluttering the output, especially the FI_WARN() output, which is more user facing than developer facing.  You might want to leave it out of the FI_WARN output.

**Comment by pmmccorm** [comment]
I think the file/line numbers should only be in FI_DEBUG output: these are only meaningful to a developer, everything else a customer/user is reading.

**Comment by pmmccorm** [comment]
Or, maybe in FI_DEBUG and in FI_LOG when log level is greater than some value ?

**Comment by goodell** [comment]
@pmmccorm I'm good with either approach.  Frankly, the macros are useful enough for me as it is, but I've enjoyed having line numbers in `FI_DEBUG`-like facilities in the past, so I don't mind the addition there.

**Comment by jithinjosepkl** [comment]
@goodell @pmmccorm - I'm leaning more towards having line number for both fi_log and fi_debug. IMHO, it can help investigating issues. We can remove it for fi_warn.

**Comment by shefty** [comment]
My preference is to have the function and line number available for log and debug.  It's trivial for an end-user to ignore the extra information, but can be helpful debugging and quickly identifying problems.  I'm more indifferent for fi_warn.  I'd like to see fi_warn output provide enough meaningful content that the user can decide whether it's relevant, and what they need to do to correct the problem.

**Comment by jithinjosepkl** [comment]
Updated the patch to remove line number info for FI_WARN

**Comment by pmmccorm** [comment]
thumbs_up.jpg

File: src/log.c [hunk 1]
```diff
@@ -69,35 +69,37 @@ void fi_warn_impl(const char *prov, const char *fmt, ...)
...
 {
 	va_list vargs;
 
 	if (prov != NULL)
-		fprintf(stderr, "%s:%s:<%d> ", PACKAGE, prov, level);
+		fprintf(stderr, "%s:%s:%s:%d<%d> ", PACKAGE, prov, 
+			func, line, level);
 	else
-		fprintf(stderr, "%s:<%d> ", PACKAGE, level);
+		fprintf(stderr, "%s:%s:%d<%d> ", PACKAGE, func, line, level);
```
Comment (shefty) [inline src/log.c]: Let's add "()" after the function name in the display.  E.g. "%s()", __func___

Comment (jithinjosepkl) [inline src/log.c]: @shefty Updated the patch accordingly.

**Pull request merged**